AGENTS.md

```diff
@@ -0,0 +1,643 @@
+
+# Run specific test within a file
+./node_modules/.bin/mocha -r "packages/dd-trace/test/setup/mocha.js" packages/dd-trace/test/appsec/sdk/track_event.spec.js --grep "should track login success"
+
+# Enable debug logging for debugging failing tests
+DD_TRACE_DEBUG=true ./node_modules/.bin/mocha -r "packages/dd-trace/test/setup/mocha.js" packages/dd-trace/test/appsec/sdk/track_event.spec.js
+```
+
+### Plugin Tests
+
+Plugin tests require external services running in Docker:
+
+```bash
+# Example for testing the amqplib plugin
+export SERVICES="rabbitmq"
+export PLUGINS="amqplib"
+
+# Start required services
+docker compose up -d $SERVICES
+
+# Install plugin versions and check services
+yarn services
+
+# Run plugin tests
+yarn test:plugins
+```
+
+To run a single plugin test file:
+
+```bash
+# Run a specific plugin test file
+./node_modules/.bin/mocha -r "packages/dd-trace/test/setup/mocha.js" packages/datadog-plugin-amqplib/test/index.spec.js
+```
+
+Check `.github/workflows/apm-integrations.yml` for the correct `SERVICES` and `PLUGINS` values for each plugin.
+
+You can test multiple plugins at once using pipe delimiters:
+
+```bash
+PLUGINS="amqplib|bluebird" yarn test:plugins
+```
+
+**Platform Limitations:** Some native modules don't compile on ARM64 devices (Apple silicon): `aerospike`, `couchbase`, `grpc`, `oracledb`. These plugin tests cannot be run locally on ARM64 devices.
+
+### Test Coverage
+
+Coverage is measured with nyc. To check coverage for your changes:
+
+1. Run tests with nyc for the component you're working on:
+   ```bash
+   ./node_modules/.bin/nyc --include "packages/dd-trace/src/debugger/**/*.js" \
+     ./node_modules/.bin/mocha -r "packages/dd-trace/test/setup/mocha.js" \
+     "packages/dd-trace/test/debugger/**/*.spec.js"
+   ```
+
+2. The output will show a table like this:
+   ```
+   ----------|---------|----------|---------|---------|-------------------
+   File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
+   ----------|---------|----------|---------|---------|-------------------
+   All files |   82.56 |    75.45 |   89.06 |   83.86 |                   
+   config.js |     100 |      100 |     100 |     100 |                   
+   ----------|---------|----------|---------|---------|-------------------
+   ```
+   - `% Stmts`: % of code statements executed
+   - `% Branch`: % of code branches (if/else) taken
+   - `% Funcs`: % of functions called
+   - `% Lines`: % of lines executed
+   - `Uncovered Line #s`: Specific line numbers not covered by tests
+
+3. Check the percentage values to ensure important code paths are covered.
+
+#### Coverage Philosophy
+
+- Don't reduce coverage with new PRs
+- Cover important production code paths unless tests become overly complex
+- Integration tests in sandboxes don't count toward metrics, so don't add redundant unit tests just for coverage numbers
```

Comment:
This line makes no sense.

To rephrase, it's saying "don't add redundant _unit_ tests, since _integration_ tests don't count". In the first part of the sentence it's suggesting that integration tests aren't used for coverage metrics, but in the second part, it's suggesting that they should be preferred. So, the implication then, is that coverage should be driven _down_. This is wrong.

Reply:
I'll completely refactor this section.